sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2344,18 +2344,36 @@ object GpuOverrides extends Logging {
         override def convertToGpu(childExprs: Seq[Expression]): GpuExpression =
           GpuSum(childExprs.head, a.dataType)
       }),
+    expr[NthValue](
+      "nth aggregate operator",
+      ExprChecks.windowOnly(
+        (TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP +
+            TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128).nested(),
+        TypeSig.all,
+        Seq(ParamCheck("input",
+          (TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP +
+              TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128).nested(),
+          TypeSig.all),
+          ParamCheck("offset", TypeSig.INT, TypeSig.INT))
```

Comment:
nit: Technically these are lit only ints. This will not change for anything we have today, but it is just being defensive.

```
ParamCheck("offset", TypeSig.lit(TypeSig.INT), TypeSig.lit(TypeSig.INT))
```